code/modules/unit_tests/closets.dm

```diff
@@ -9,10 +9,11 @@
 
 	for(var/closet_type in all_closets)
 		var/obj/structure/closet/closet = allocate(closet_type)
+		if (QDELETED(closet)) // this is here because the emcloset subtype has a chance of returning a qdel hint on initialize
+			continue
```

Comment:
```suggestion
	if((closet_type == /obj/structure/closet/emcloset)) && QDELETED(closet))
		continue
```

Would feel much better about this being an explicit check just incase something unintended deletes in the future

Reply:
Good point